### ISSUE TITLE
Installation Error: No Matching Version Found for @tanstack/virtual-core@3.10.0

### ISSUE THREAD

**Issue by wilsuriel03** [tailwindlabs/headlessui#3430]
**What package within Headless UI are you using?**

`@headlessui/react`

**What version of that package are you using?**

`v2.1.2`

**What browser are you using?**

N/A (Issue occurs during package installation)

**Reproduction URL**

Since this is an issue with package installation, a CodeSandbox reproduction is not applicable. However, the issue can be reproduced with the following steps:

1. Create a new project or navigate to an existing project.
2. Attempt to install **@headlessui/react@2.1.2** using pnpm with the command:
 `pnpm add @headlessui/react@latest`
4. Observe the error related to the missing version of` @tanstack/virtual-core.`

**Describe your issue**

When trying to install the latest version of `@headlessui/react@2.1.2` using pnpm, the installation fails with the following error message:

```js
 ERR_PNPM_NO_MATCHING_VERSION  No matching version found for @tanstack/virtual-core@3.10.0

This error happened while installing the dependencies of @headlessui/react@2.1.2
 at @tanstack/react-virtual@3.10.0

The latest release of @tanstack/virtual-core is "3.9.0".
```

It appears that the dependency `@tanstack/react-virtual@3.10.0 `is expecting `@tanstack/virtual-core@3.10.0`, which does not exist in the registry. The latest available version of `@tanstack/virtual-core is 3.9.0.`

This issue prevents the installation of the `@headlessui/react` package, blocking usage in projects that rely on it.

**Comment by beeirl** [comment]
same issue here

**Comment by Lawliang** [comment]
same

**Comment by AlissonRoss** [comment]
Same... rollback to 2.0

**Comment by Lawliang** [comment]
> If you're experiencing this issue, try rolling back to `@headlessui/react@2.1.1`, which works without the dependency problem.

did this work for everyone? i'm still getting the dependency error after rolling back to 2.1.1

**Comment by bigshirtjonny** [comment]
> did this work for everyone? i'm still getting the dependency error after rolling back to 2.1.1

@Lawliang - Yep this worked for me

**Comment by THAgility** [comment]
Version 2.1.1 worked for me

**Comment by nilsbunger** [comment]
The underlying issue in tanstack is tracked [here](https://github.com/TanStack/virtual/issues/804)

**Comment by owenherbert** [comment]
I fixed my deployment pipelines by changing my dependency to `"@headlessui/react": "2.1.1"`

**Comment by Yuripetusko** [comment]
Fixed by adding this to package.json

```
"resolutions": {
    "@tanstack/virtual-core": "3.9.0"
  }
```

**Comment by varzoeaa** [comment]
thank you so much! it works for me too <3

**Comment by manuelMoraYolo** [comment]
Same for me

**Comment by DominikButz** [comment]
same, just installed 2.0.0-alpha.4 instead.

**Comment by e1016** [comment]
solved upgrading to `"@headlessui/react": "^2.1.2"`

**Comment by RobinMalfait** [comment]
Hey!

We don't pin an exact version of `@tanstack/virtual`, so no change is required from our end. They made a small bug and published a version and unpublished it.

This should all be solved already 👍
